cmd/sync/main.go

```diff
@@ -189,3 +192,49 @@ func getStreamUpstreamServerAddresses(server []nginx.StreamUpstreamServer) []str
 	}
 	return streamUpstreamServerAddr
 }
+
+// headerTransport wraps an http.RoundTripper and adds custom headers to all requests.
+type headerTransport struct {
+	headers   http.Header
+	transport http.RoundTripper
+}
+
+func (t *headerTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(t.headers)+len(req.Header) > maxHeaders {
+		return nil, fmt.Errorf("number of headers in request exceeds the maximum allowed (%d)", maxHeaders)
+	}
+	clonedReq := req.Clone(req.Context())
+
+	for key, values := range t.headers {
+		for _, value := range values {
+			clonedReq.Header.Add(key, value)
+		}
+	}
+
+	resp, err := t.transport.RoundTrip(clonedReq)
+	if err != nil {
+		return nil, fmt.Errorf("headerTransport RoundTrip failed: %w", err)
+	}
+
+	return resp, nil
```

Comment:
we're not really adding any new information in our `fmt.Errorf`, let's skip it
```suggestion
	return t.transport.RoundTrip(clonedReq)
```